### ISSUE TITLE
PDF is white in Xournal++

### ISSUE THREAD

**Issue by efaulhaber** [xournalpp/xournalpp#6167]
### Operating System

Mac OS

### (Linux Only) Display Server

None

### Installation Method

Homebrew

### Xournal++ Version

1.2.5

### libgtk Version

3.24.42

### Bug Description

When I open the attached PDF, it just shows a white page. I see this with several other scanned documents as well.

[Theory Blatt 8.pdf](https://github.com/user-attachments/files/18195159/Theory.Blatt.8.pdf)

### Expected Behaviour

I expect to see the content of the PDF.

### Steps to Reproduce

Open attached PDF in Xournal++. Admire the white page.

**Comment by efaulhaber** [comment]
I had my colleague test this on Linux and he can see the contents, so it seems this is a macOS bug?

**Comment by rolandlo** [comment]
I can reproduce on MacOS (both Intel and ARM). There is an error message:
```term
Fontconfig error: Cannot load default config file: No such file: (null)
```

**Comment by rolandlo** [comment]
> I can reproduce on MacOS (both Intel and ARM). There is an error message:
> 
> ```
> Fontconfig error: Cannot load default config file: No such file: (null)
> ```

The error message stems from the issue that `fontconfig` can't find its default config file. It can be set using the environment variable `FONTCONFIG_PATH`. On Linux that would be `/etc/fonts`. On MacOS we probably would need to bundle those files as well through `xournalpp.bundle`. However we don't do that on Windows either and there the PDF looks fine in Xournal++. So there is likely a different cause.

**Comment by schlomo** [comment]
I have the same problem, also with the recent nightly:
```
❯ /Applications/Xournal++.app/Contents/MacOS/xournalpp --version
xopp-Message: 08:06:41.197: Continue with GDK_PIXBUF_MODULE_FILE = /Applications/Xournal++.app/Contents/Resources/lib/gdk-pixbuf-2.0/2.10.0/loaders.cache
GI_TYPELIB_PATH = /Applications/Xournal++.app/Contents/Resources/lib/girepository-1.0
DYLD_FALLBACK_LIBRARY_PATH = (null)
LUA_PATH = /Applications/Xournal++.app/Contents/Resources/share/lua/5.4/?.lua
LUA_CPATH = /Applications/Xournal++.app/Contents/Resources/lib/lua/5.4/?.so
xopp-Message: 08:06:41.199: Setting LANG and LC_MESSAGES to en_IE.UTF-8
xournalpp 1.2.5+dev (99228da3)
└──libgtk: 3.24.42
```

How can we help you to fix this?

**Comment by rolandlo** [comment]
In version 1.2.4 it still worked, in 1.2.5 it doesn't. Here is some potentially relevant file diff (where Xournal++ 6.app is 1.2.4):
```term
Only in Downloads/Xournal++ 6.app/Contents/Resources/lib/gdk-pixbuf-2.0/2.10.0/loaders: libpixbufloader-ani.so
Only in Downloads/Xournal++ 6.app/Contents/Resources/lib/gdk-pixbuf-2.0/2.10.0/loaders: libpixbufloader-bmp.so
Only in Downloads/Xournal++ 6.app/Contents/Resources/lib/gdk-pixbuf-2.0/2.10.0/loaders: libpixbufloader-icns.so
Only in Downloads/Xournal++ 6.app/Contents/Resources/lib/gdk-pixbuf-2.0/2.10.0/loaders: libpixbufloader-ico.so
Only in Downloads/Xournal++ 6.app/Contents/Resources/lib/gdk-pixbuf-2.0/2.10.0/loaders: libpixbufloader-pnm.so
Only in Downloads/Xournal++ 6.app/Contents/Resources/lib/gdk-pixbuf-2.0/2.10.0/loaders: libpixbufloader-qtif.so
Only in Downloads/Xournal++ 6.app/Contents/Resources/lib/gdk-pixbuf-2.0/2.10.0/loaders: libpixbufloader-tga.so
Only in Downloads/Xournal++ 6.app/Contents/Resources/lib/gdk-pixbuf-2.0/2.10.0/loaders: libpixbufloader-xbm.so
Only in Downloads/Xournal++ 6.app/Contents/Resources/lib/gdk-pixbuf-2.0/2.10.0/loaders: libpixbufloader-xpm.so
Only in /Applications/Xournal++.app/Contents/Resources/lib: libharfbuzz-gobject.0.dylib
Only in /Applications/Xournal++.app/Contents/Resources/lib: libiconv.2.dylib
Only in /Applications/Xournal++.app/Contents/Resources/lib: libjpeg.62.dylib
Only in Downloads/Xournal++ 6.app/Contents/Resources/lib: libjpeg.9.dylib
```

**Comment by schlomo** [comment]
[Xournal Test File.pdf](https://github.com/user-attachments/files/18330852/Xournal.Test.File.pdf)

Here ist a test PDF that exhibits the problem for me. Feel free to add it to a test suite or such.

**Comment by rolandlo** [comment]
> [Xournal Test File.pdf](https://github.com/user-attachments/files/18330852/Xournal.Test.File.pdf)
> 
> Here ist a test PDF that exhibits the problem for me. Feel free to add it to a test suite or such.

Thanks for the message. :smile:  Working on a fix.

**Comment by jean2q** [comment]
I encountered the issue on version 1.2.5, so I uninstalled it and reverted to the previous version (1.2.4).
It seems to work fine on my MacBook Air M1.
Thanks everyone.

**Comment by rolandlo** [comment]
This is fixed now in the release branch (so a version 1.2.6 will have the fix). Once the release branch is merged into the master branch it will be fixed in the nightly builds as well.
I'm closing the issue.
